### DISCUSSION TITLE
Assets, credits and storage

### DISCUSSION THREAD

**Discussion by erlend-sh** [fishfight/punchy#231] [Ideas]
This conversation is relevant to all @spicylobstergames, so solving it properly for Punchy will solve it for the rest as well.

Our artists communicate primarily on Discord, and both their WIP and final assets are typically shared directly on Discord. For lossless .png files it’s a pretty convenient form of file transfer. 

However, the lack of structure in the chat medium makes us sometimes lose track of completed assets, which is Not Great ™️ 

We’re gonna try clean up our act before this becomes a bigger issue.

## Shared assets drive

In this shared drive we’re keeping all Fish Folk media assets:

https://drive.google.com/drive/folders/1-92OjEYxFzJ1qiH3xR9xP4J1bqk-rvnC

In regular intervals (probably coinciding with game releases that introduce asset updates) we will upload this folder to our asset pack on itch.io:
https://spicylobster.itch.io/fish-world-pack

## Credits conventions 

Jumpy has a [CREDITS.md](https://github.com/fishfolks/jumpy/blob/main/CREDITS.md) file and an [in-game screen](https://github.com/fishfolks/jumpy/issues/304) that re-renders this info in-game. The data is currently transcribed across by hand.

I think we will start using our shared Drive as our source of truth for the `CREDITS.md` file. 

In this file, in a second section below the basic credits page just listing names, we should have a complete list (maybe table) of which assets came from who.

But we can crowdsource this. As long as all assets are listed, asset authors will edit in their own attributions accordingly. In the case of inactive contributors, the rest of us will attribute them to the best of our ability.

## Automation

We can incrementally glue these steps closer together, and introduce automation that enforces the basic conventions.

It would for instance be pretty cool if we had a tool that recognized when there are files in our Drive that don’t yet exist on our GitHub repo.

**Comment by zicklag** [comment]
If we add a `CREDITS.md` file I can setup the game to automatically generate the credits page from the markdown file.

And eventually it should be possible to setup a GitHub action that would run every other day or something like that and check that the `CREDITS.md` file and assets are up-to-date with the drive, and we can probably have it automatically make a pull request or, potentially more conveniently, just make a commit, to update.

**Comment by zicklag** [comment]
I'm not sure it would be a good option or not, but I have been experimenting with [Directus](https://directus.io), which is like frontend for databases that lets you store files and other custom data, add custom fields to your data, organize, display, and search that data, etc. And it provides an API so you can get all that data programmatically.

Anyway, we might be able to use it to organize our assets. For instance, we can allow all of our asset developers to log in with either GitHub or Google or Facebook, and they will be able to upload files. We could also add a contributors field to each file that we could use to track contributing authors and that we could reference when automatically generating a detailed credits listing.

We might also be able to get a Directus cloud instance for free because we're an Open Source project ( we'd have to double-check because our assets are necessarily totally open source ).

I've never used Google drive so I'm not sure how that compares, and I'm not really sure whether or not it's worth trying out or not, but I figured I'd throw it out there.
